### PR TITLE
tomboy-ng: init at 0.40

### LINKED ISSUE #318021: Package request: tomboy-ng
**Project description**

tomboy-ng is a note taking app that works and syncronises between Linux, Windows and MacOS. It will also Sync to a private Github repository allowing access to any device with a browser. It features a rich text markup, printing, spell check, backup and snapshot capability. Undo and Redo. Import and export (plain text, RFT, MarkDown). It has Tomboy's automatic linking between notes, searching abilities, NoteBooks and a similar interface. Change a note's title by just editing the title in the edit window.

**Metadata**

* homepage URL: https://github.com/tomboy-notes/tomboy-ng
* source URL: https://github.com/tomboy-notes/tomboy-ng/tree/master/source
* license: MIT License
* platforms:  linux

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

### PR DESCRIPTION
**Pull request by pluiedev**
## Description of changes

Fixes #318021

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/to/tomboy-ng/package.nix [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+    };
+    kcontrols = fetchFromGitHub {
+      owner = "davidbannon";
+      repo = "KControls";
+      rev = "4b74f50599544aa05d76385c21795ca9026e9657";
+      hash = "sha256-AHpcbt5v9Y/YG9MZ/zCLLH1Pfryv0zH8UFCgY/RqrdQ=";
+      name = "kcontrols";
+    };
+  in
+  {
```
Comment (getchoo) [inline pkgs/by-name/to/tomboy-ng/package.nix:37]:
```suggestion
stdenv.mkDerivation (finalAttrs: {
```

File: pkgs/by-name/to/tomboy-ng/package.nix [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+    pname = "tomboy-ng";
+    version = "0.39";
+
+    srcs = [
+      src
+      kcontrols
+    ];
```
Comment (getchoo) [inline pkgs/by-name/to/tomboy-ng/package.nix:44]:
```suggestion
    src = fetchFromGitHub {
      owner = "tomboy-notes";
      repo = "tomboy-ng";
      rev = "v${finalAttrs.version}";
      hash = "sha256-2zU61/ulFJa1337SqbUQFi/Hv5s+wM3h3t/nIR/3XvM=";
    };
```

File: pkgs/by-name/to/tomboy-ng/package.nix [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+
+    sourceRoot = src.name;
+
```
Comment (getchoo) [inline pkgs/by-name/to/tomboy-ng/package.nix:47]:
```suggestion
```

File: pkgs/by-name/to/tomboy-ng/package.nix [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+      gtk2
+      gdk-pixbuf
+      at-spi2-atk
+      xorg.libX11
+      libnotify
+    ];
+
+    patches = [ ./simplify-build-script.patch ];
+
+    preBuild = "ln -s ../kcontrols kcontrols";
```
Comment (getchoo) [inline pkgs/by-name/to/tomboy-ng/package.nix:67]:
```suggestion
    preBuild = "ln -s ${finalAttrs.passthru.kcontrols} kcontrols";
```

File: pkgs/by-name/to/tomboy-ng/package.nix [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+
+    makeFlags = [ "PREFIX=${placeholder "out"}" ];
+
+    passthru.updateScript = nix-update-script { };
```
Comment (getchoo) [inline pkgs/by-name/to/tomboy-ng/package.nix:71]:
```suggestion
    passthru = {
      updateScript = nix-update-script { };
      kcontrols = fetchFromGitHub {
        owner = "davidbannon";
        repo = "KControls";
        rev = "4b74f50599544aa05d76385c21795ca9026e9657";
        hash = "sha256-AHpcbt5v9Y/YG9MZ/zCLLH1Pfryv0zH8UFCgY/RqrdQ=";
      };
    };
```

**Review by getchoo — CHANGES_REQUESTED** [review]

File: pkgs/by-name/to/tomboy-ng/package.nix (hunk 5 above)
Comment (Scrumplex) [reply pkgs/by-name/to/tomboy-ng/package.nix:71]: There is no reason to include `kcontrols` in `passthru`, as it is directly used by the derivation. I think it should just be a regular attribute of the derivation.

File: pkgs/by-name/to/tomboy-ng/package.nix [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+  pname = "tomboy-ng";
+  version = "0.39";
+
+  src = fetchFromGitHub {
+    owner = "tomboy-notes";
+    repo = "tomboy-ng";
+    rev = "v${finalAttrs.version}";
+    hash = "sha256-2zU61/ulFJa1337SqbUQFi/Hv5s+wM3h3t/nIR/3XvM=";
+  };
+  kcontrols = fetchFromGitHub {
```
Comment (SuperSandro2000) [inline pkgs/by-name/to/tomboy-ng/package.nix:30]: Please move this into a let in

File: pkgs/by-name/to/tomboy-ng/package.nix [hunk 7]
```diff
@@ -0,0 +1,79 @@
...
+    gtk2
+    gdk-pixbuf
+    at-spi2-atk
+    xorg.libX11
+    libnotify
+  ];
+
+  patches = [ ./simplify-build-script.patch ];
+
+  preBuild = "ln -s ${finalAttrs.kcontrols} kcontrols";
```
Comment (SuperSandro2000) [inline pkgs/by-name/to/tomboy-ng/package.nix:57]: This should be fine in postPatch

File: pkgs/by-name/to/tomboy-ng/package.nix (hunk 6 above)
Comment (Scrumplex) [reply pkgs/by-name/to/tomboy-ng/package.nix:30]: Wouldn't that make it difficult to override this source derivation?

**Review by Scrumplex — COMMENTED** [review]
needs nixfmt

**Review by Scrumplex — APPROVED** [review]
Builds and runs.. kinda. I had some odd behavior but I might just not be using it correctly.

**Pull request merged**